src/bundles/analytics.test.js

```diff
@@ -27,152 +27,169 @@ function createStore (analyticsOpts = {}) {
+    expect(store.selectShowAnalyticsBanner()).toBe(false)
+  })
 })
 
-it('should enable analytics if user has explicitly enabled it', () => {
-  const store = createStore()
-  store.doEnableAnalytics()
-  expect(store.selectAnalyticsEnabled()).toBe(true)
-  expect(store.selectAnalyticsConsent()).toEqual(['sessions', 'events', 'views', 'location'])
-  expect(global.Countly.opt_in).toHaveBeenCalled()
-  expect(global.Countly.opt_out).not.toHaveBeenCalled()
-  expect(global.Countly.opt_in.mock.calls.length).toBe(1)
+describe('user enables and disables analytics', () => {
+  it('should enable analytics if user who has opted out explicitly enables it', () => {
+    const mockDefaultState = {
+      lastEnabledAt: 0,
+      lastDisabledAt: Date.now(),
+      showAnalyticsBanner: false,
+      optedOut: true,
+      consent: []
+    }
+    const store = createStore({}, mockDefaultState)
+    store.doEnableAnalytics()
+    expect(store.selectAnalyticsEnabled()).toBe(true)
+    expect(store.selectAnalyticsConsent()).toEqual(['sessions', 'events', 'views', 'location'])
+    expect(global.Countly.opt_in).toHaveBeenCalled()
+    expect(global.Countly.opt_out).not.toHaveBeenCalled()
+    expect(global.Countly.opt_in.mock.calls.length).toBe(1)
+  })
+
+  it('should disable analytics if user explicitly disables it', () => {
+    const store = createStore()
+    store.doDisableAnalytics()
+    expect(store.selectAnalyticsEnabled()).toBe(false)
+    expect(store.selectAnalyticsConsent()).toEqual([])
+    expect(global.Countly.opt_out).toHaveBeenCalled()
+    expect(global.Countly.opt_out.mock.calls.length).toBe(1)
+  })
 })
 
-it('should disable analytics if user has explicitly disabled it', () => {
-  const store = createStore()
-  store.doDisableAnalytics()
-  expect(store.selectAnalyticsEnabled()).toBe(false)
-  expect(store.selectAnalyticsConsent()).toEqual([])
-  expect(global.Countly.opt_in).not.toHaveBeenCalled()
-  expect(global.Countly.opt_out).toHaveBeenCalled()
-  expect(global.Countly.opt_out.mock.calls.length).toBe(1)
+describe('user manages all analytics consent with settings page toggle', () => {
+  it('should toggle analytics consent off for user with consent enabled', () => {
+    const mockDefaultState = {
+      lastEnabledAt: Date.now(),
+      lastDisabledAt: 0,
+      consent: ['sessions', 'events', 'views'],
+      optedOut: false
+    }
+    const store = createStore({}, mockDefaultState)
+    store.doToggleAnalytics()
+    expect(store.selectAnalyticsEnabled()).toBe(false)
+    expect(store.selectAnalyticsConsent()).toEqual([])
+    expect(global.Countly.opt_in).not.toHaveBeenCalled()
+    expect(global.Countly.opt_out).toHaveBeenCalled()
+    expect(global.Countly.opt_out.mock.calls.length).toBe(1)
+  })
+  it('should toggle analytics consent on for user with consent disabled', () => {
+    const mockDefaultState = {
+      lastEnabledAt: 1674166495717, // past
+      lastDisabledAt: Date.now(),
```

Comment:
make it easy to update, and see what the actual dates are.
```suggestion
      lastEnabledAt: (new Date('2022-01-01')).getTime(),
      lastDisabledAt: (new Date('2022-01-02')).getTime(),
```